diff --git a/internal/ceres/lapack.cc b/internal/ceres/lapack.cc
index 37efbcd..cf1c948 100644
--- a/internal/ceres/lapack.cc
+++ b/internal/ceres/lapack.cc
@@ -36,31 +36,31 @@
 
 #ifndef CERES_NO_LAPACK
 // C interface to the LAPACK Cholesky factorization and triangular solve.
-extern "C" void dpotrf_(char* uplo,
-                       int* n,
+extern "C" void dpotrf_(const char* uplo,
+                       const int* n,
                        double* a,
-                       int* lda,
+                       const int* lda,
                        int* info);
 
-extern "C" void dpotrs_(char* uplo,
-                        int* n,
-                        int* nrhs,
-                        double* a,
-                        int* lda,
+extern "C" void dpotrs_(const char* uplo,
+                        const int* n,
+                        const int* nrhs,
+                        const double* a,
+                        const int* lda,
                         double* b,
-                        int* ldb,
+                        const int* ldb,
                         int* info);
 
-extern "C" void dgels_(char* uplo,
-                       int* m,
-                       int* n,
-                       int* nrhs,
+extern "C" void dgels_(const char* uplo,
+                       const int* m,
+                       const int* n,
+                       const int* nrhs,
                        double* a,
-                       int* lda,
+                       const int* lda,
                        double* b,
-                       int* ldb,
+                       const int* ldb,
                        double* work,
-                       int* lwork,
+                       const int* lwork,
                        int* info);
 #endif
 
