diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf309e9..d36ba3c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -644,7 +644,7 @@ INSTALL(PROGRAMS ${NCXX_BINARY_DIR}/ncxx4-config
 #####
 IF(NC_HAS_DEF_VAR_FILTER)
   SET(ISCMAKE "1")
-  configure_file(${CMAKE_SOURCE_DIR}/cxx4/findplugin.in ${CMAKE_BINARY_DIR}/cxx4/findplugin.sh @ONLY NEWLINE_STYLE LF)
+  configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cxx4/findplugin.in ${CMAKE_CURRENT_BINARY_DIR}/cxx4/findplugin.sh @ONLY NEWLINE_STYLE LF)
 ENDIF(NC_HAS_DEF_VAR_FILTER)
 #####
 # End findplugin.sh
@@ -645,7 +645,7 @@ IF(NC_HAS_DEF_VAR_FILTER)
   # Build cxx4/findplugin.sh
   #####
   SET(ISCMAKE "1")
-  CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/cxx4/findplugin.in ${CMAKE_BINARY_DIR}/nc_test4/findplugin.sh @ONLY NEWLINE_STYLE LF)
+  CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/cxx4/findplugin.in ${CMAKE_CURRENT_BINARY_DIR}/nc_test4/findplugin.sh @ONLY NEWLINE_STYLE LF)
 
   ADD_SUBDIRECTORY(plugins)
 ENDIF(NC_HAS_DEF_VAR_FILTER)
